lib/LWP/Protocol/Net/Curl.pm

```diff
@@ -217,6 +217,18 @@ sub _handle_method {
         },
     );
 
+	if (my $protocol = $request->protocol) {
+		my $v;
+		if ($protocol =~ m/^HTTP\/1\.0$/) {
+			$v = CURL_HTTP_VERSION_1_0;
+		} elsif ($protocol =~ m/^HTTP\/1\.1$/) {
+			$v = CURL_HTTP_VERSION_1_1;
+		} elsif ($protocol =~ m/^HTTP\/2$/) {
```

Comment:
Are you sure it's always `HTTP/2` and never `HTTP/2.0`? The RFC has at least one occurrence of the latter: https://tools.ietf.org/html/rfc7540#section-3.5

Reply:
Also, libcurl is much more granular regarding the version setting; it even has HTTP/3: https://github.com/curl/curl/blob/master/docs/libcurl/symbols-in-versions#L832-L839